### ISSUE TITLE
Streamlining exit on error procedures

### ISSUE THREAD

**Issue by fouriaux** [BlueBrain/CoreNeuron#95]
create_report_configuration can fail and raise error using abort(), this lead to an error not necessary easy to interpret from a normal user without a debugger.

this abort and potentially other ones must be replaced by proper error management.

**Comment by tapaswenipathak** [comment]
Hi @fouriaux, if you have branch let me know otherwise I will PR.

**Comment by pramodk** [comment]
@tapaswenipathak : I dont think we started anything on this. But I would keep this in queue for time being as we are going to change overall execution workflow soon.

**Comment by tapaswenipathak** [comment]
Hi @pramodk, can I work on this or is this internal? I can create a draft PR for proper error management?

**Comment by pramodk** [comment]
@tapaswenipathak : Sure. We should consider parallel aspects while error handling (i.e. MPI executions). I haven't thought about this in detail but if you already thinking about this / some abstraction, let me know how.

**Comment by tapaswenipathak** [comment]
@pramodk Can I go ahead and implement https://github.com/gabime/spdlog? Do you have any package in mind which would work w/ MPI executions?

**Comment by tapaswenipathak** [comment]
I just saw `coreneuron/scopmath_core/abort.cpp` file but `abort_run` is being used at very few places, should I implement that completely?

**Comment by tapaswenipathak** [comment]
ping @pramodk, do you think I can enhance the PR? https://github.com/BlueBrain/CoreNeuron/pull/171, please let me know.

**Comment by olupton** [comment]
See also: #316.
